lib/src/content/components/main_content/wolt_modal_sheet_top_bar_title.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:wolt_modal_sheet/src/modal_page/wolt_modal_sheet_page.dart';
+
+/// A widget to display the top bar title in a modal sheet.
+///
+/// It tries to display the title in the following order of preference:
+/// - The `topBarTitle` widget if it's not null.
+/// - The text data in `pageTitle` if it's a `Text` widget.
+/// - The first text descendant of the `pageTitle` if it's not a `Text` widget and has a
+///   [Text] descendant.
+/// - If none of the above are applicable, it defaults to a `SizedBox.shrink`.
+class WoltModalSheetTopBarTitle extends StatefulWidget {
+  const WoltModalSheetTopBarTitle({
+    Key? key,
+    required this.page,
+    required this.pageTitleKey,
+  }) : super(key: key);
+
+  final WoltModalSheetPage page;
+  final GlobalKey pageTitleKey;
+
+  @override
+  State<WoltModalSheetTopBarTitle> createState() => _WoltModalSheetTopBarTitleState();
+}
+
+class _WoltModalSheetTopBarTitleState extends State<WoltModalSheetTopBarTitle> {
+  String? pageTitleText;
+
+  @override
+  void initState() {
+    super.initState();
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      _extractPageTitleText();
+    });
+  }
+
+  void _extractPageTitleText() {
+    final isPageTitleTextWidget = widget.page.pageTitle is Text;
+    final pageTitleElement = widget.pageTitleKey.currentContext;
+
+    if (widget.page.topBarTitle != null || isPageTitleTextWidget) {
+      return;
+    } else if (pageTitleElement != null) {
+      _visitAllDescendants(pageTitleElement);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final topBarTitle = widget.page.topBarTitle;
+    final pageTitleText =
+        (widget.page.pageTitle is Text) ? (widget.page.pageTitle as Text).data : this.pageTitleText;
+
+    if (topBarTitle != null) {
+      return topBarTitle;
+    } else if (pageTitleText != null) {
+      return Row(
+        children: [
+          const SizedBox(width: 72),
+          Expanded(
+            child: Text(
+              pageTitleText,
+              style: Theme.of(context).textTheme.titleSmall,
+              textAlign: TextAlign.center,
+              overflow: TextOverflow.ellipsis,
+              maxLines: 1,
+            ),
+          ),
+          const SizedBox(width: 72),
+        ],
+      );
+    }
+    return const SizedBox.shrink();
```

Comment:
This might be fine as an initial implementation but this needs to be improved. 
I would recommend adding a TODO here and an issue in the repository to improve this later.

Reply:
@TahaTesser can you please mention how it can be improved?

Reply:
Using a custom child layout determine the element before the layout happens. Also, it returns a sized box, this widget doesn't render at all if there are no conditions.

The current look-up is only in the `init`, so what happens if the title text widget is removed? Does it update the widget?  Text is in inside `Expanded` which will render the widget too wide. It should alignment with the text direction and ability to customize padding. Lower level widget like `NavigationToolbar` already has these features.

Reply:
@TahaTesser Thanks, in my next PR, I will apply your suggestions. Merging this PR for now.